zetteldeft.el

```diff
@@ -240,6 +240,38 @@ This is done with the regular expression stored in
     (when (re-search-forward zetteldeft-id-regex nil t -1)
       (match-string 0))))
 
+(defun zetteldeft--insert-link (id &optional title)
+  "Insert a link to Zetteldeft note ID.
+If TITLE is included, use it as link text."
+  (interactive)
+  (funcall zetteldeft-insert-link-function id title))
+
+(defcustom zetteldeft-insert-link-function
+           #'zetteldeft--insert-link-zd-style
+  "The function to use when inserting note links.
+
+Use either
+ - `zetteldeft--insert-link-zd-style' for Zetteldeft type links
+ - `zetteldeft--insert-link-org-style' for Org-mode zdlink: links
+ - A custom function that takes two arguments: an ID and an optional title."
+  :type 'function
+  :options '(zetteldeft--insert-link-zd-style
+             zetteldeft--insert-link-org-style)
+  :group 'zetteldeft)
+
+(defun zetteldeft--insert-link-zd-style (id &optional title)
+  "Insert a Zetteldeft link to note with provided ID."
+  (insert zetteldeft-link-indicator
+          id
+          zetteldeft-link-suffix)
+  (when title (insert " " title)))
+
+(defun zetteldeft--insert-link-org-style (id &optional title)
+  "Insert a Zetteldeft link in Org-mode format as zdlink: type."
+  (if title
+      (insert "[[zdlink:" id "][" title "]")
```

Comment:
There's a small typo here - should be:

```lisp
(insert "[[zdlink:" id "][" title "]]")
```

Reply:
Thanks, fixed. Hadn't tested this yet. I might also have to include some more details on how to set up these `zdlink:` type links